frontend/public/components/namespace.jsx

```diff
@@ -210,49 +209,64 @@ const namespaceDropdownStateToProps = state => {
   return { activeNamespace, canListNS };
 };
 
-const NamespaceDropdown = connect(namespaceDropdownStateToProps)(props => {
-  const { activeNamespace, dispatch, canListNS, useProjects } = props;
-  if (canListNS === undefined) {
-    return null;
-  }
 
-  const { loaded, data } = props.namespace;
-  const model = getModel(useProjects);
-  const allNamespacesTitle = `all ${model.labelPlural.toLowerCase()}`;
-  const items = {};
-  if (canListNS) {
-    items[ALL_NAMESPACES_KEY] = allNamespacesTitle;
+class NamespaceDropdown_ extends React.Component {
+
+  componentDidUpdate() {
+    const { namespace, dispatch } = this.props;
+    if (namespace.loaded) {
+      const projectsAvailable = !_.isEmpty(namespace.data);
+      setFlag(dispatch, FLAGS.PROJECTS_AVAILABLE, projectsAvailable);
+    }
   }
-  _.map(data, 'metadata.name').sort().forEach(name => items[name] = name);
-
-  let title = activeNamespace;
-  if (activeNamespace === ALL_NAMESPACES_KEY) {
-    title = allNamespacesTitle;
-  } else if (loaded && !_.has(items, title)) {
-    // If the currently active namespace is not found in the list of all namespaces, put it in anyway
-    items[title] = title;
+
+  render() {
+    const { activeNamespace, dispatch, canListNS, useProjects } = this.props;
+    if (canListNS === undefined) {
+      return null;
+    }
+
+    const { loaded, data } = this.props.namespace;
+    const model = getModel(useProjects);
+    const allNamespacesTitle = `all ${model.labelPlural.toLowerCase()}`;
+    const items = {};
+    if (canListNS) {
+      items[ALL_NAMESPACES_KEY] = allNamespacesTitle;
+    }
+
+    _.map(data, 'metadata.name').sort().forEach(name => items[name] = name);
```

Comment:
Yeah, I do agree that this could be refactored, but this was existing code, so I didn't revisit it as part of this PR. Everything in the `render` method was existing. I just wrapped this in a class so that I could use the `componentDidUpdate` function to set the `PROJECTS_AVAILABLE` flag.

Reply:
Got it, no refacor needed now then. thx!